rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java

```diff
@@ -1134,9 +1108,9 @@ public String eth_newFilter(FilterRequest fr) throws Exception {
 
             if (fr.address instanceof String) {
                 logFilter.withContractAddress(stringHexToByteArray((String) fr.address));
-            } else if (fr.address instanceof String[]) {
+            } else if (fr.address instanceof List<?> && !((List)fr.address).isEmpty() && ((List)fr.address).get(0) instanceof String) {
```

Comment:
I think it would be better to allow an instance of any `Collection`, and also filter to only allow objects that are an instance of the `String` class. This might be an implementation:

```java
            } else if (fr.address instanceof Collection<?>) {
                Collection<?> iterable = (Collection<?>) fr.address;
                byte[][] addr = iterable.stream()
                        .filter(String.class::isInstance).map(String.class::cast)
                        .map(TypeConverter::stringHexToByteArray)
                        .toArray(byte[][]::new);
                logFilter.withContractAddress(addr);
            }
```